src/pkg/packager/deploy.go

```diff
@@ -377,19 +377,34 @@ func (p *Packager) pushImagesToRegistry(componentImages []string, noImgChecksum
 
 // Push all of the components git repos to the configured git server
 func (p *Packager) pushReposToRepository(reposPath string, repos []string) error {
-	// Try repo push up to 3 times
 	for _, repoURL := range repos {
-		gitClient := git.New(p.cfg.State.GitServer)
-		repoPath, err := gitClient.TransformURLtoRepoName(repoURL)
-		if err != nil {
-			return fmt.Errorf("unable to get the repo name from the URL %s: %w", repoURL, err)
+
+		// Create an anonymous function to push the repo to the Zarf git server
+		tryPush := func() error {
+			gitClient := git.New(p.cfg.State.GitServer)
+
+			// If this is a serviceURL, create a port-forward tunnel to that resource
+			if cluster.IsServiceURL(gitClient.Server.Address) {
+				if tunnel, err := cluster.NewTunnelFromServiceURL(gitClient.Server.Address); err != nil {
```

Comment:
It's a bit in the nerd-weeds but since you brought it up earlier, I don't actually believe there will be a GC issues with this setup. Brushed up on some golang GC behavior and I _think_ because there will be a goroutine that is still accessing the pointer internally that the value will not be GCd early.